paddle/fluid/operators/conv_mkldnn_op.cc

```diff
@@ -131,21 +167,198 @@ class ConvMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
+      }
+      if (!is_INT8) {
+        if (fuse_residual_conn) {
+          auto residual_param = ctx.Input<Tensor>("ResidualData");
+          auto residual_param_data = residual_param->data<T>();
+          if (residual_param->format() != handler->GetDstFormat()) {
+            auto output_data = output->mutable_data<T>(
+                ctx.GetPlace(), ::paddle::memory::Allocator::kDefault,
+                handler->GetDstMemorySize());
+            auto residual_data_tz =
+                paddle::framework::vectorize2int(residual_param->dims());
+            auto residual_data_type =
+                paddle::framework::ToMKLDNNDataType(residual_param->type());
+
+            auto user_residual_md = platform::MKLDNNMemDesc(
+                residual_data_tz, residual_data_type, residual_param->format());
+            auto user_residual_memory_p = handler->AcquireResidualDataMemory(
+                user_residual_md, to_void_cast<T>(residual_param_data));
+            dst_memory_p = handler->AcquireDstMemoryFromResidualDataMemory(
+                user_residual_memory_p, to_void_cast<T>(output_data), pipeline);
+          } else {
+            output->ShareDataWith(*residual_param);
+            auto output_data = output->mutable_data<T>(ctx.GetPlace());
+            dst_memory_p->set_data_handle(to_void_cast<T>(output_data));
+          }
+        } else {
+          auto output_data = output->mutable_data<T>(
+              ctx.GetPlace(), ::paddle::memory::Allocator::kDefault,
+              handler->GetDstMemorySize());
+          dst_memory_p->set_data_handle(to_void_cast<T>(output_data));
+        }
+      } else if (is_INT8) {
+        if (fuse_residual_conn) {
+          auto residual_param = ctx.Input<Tensor>("ResidualData");
+          auto residual_dt =
+              paddle::framework::ToMKLDNNDataType(residual_param->type());
+          output->ShareDataWith(*residual_param);
+          if (residual_dt == mkldnn::memory::data_type::u8) {
+            uint8_t* output_data =
+                output->mutable_data<uint8_t>(ctx.GetPlace());
+            dst_memory_p->set_data_handle(to_void_cast<uint8_t>(output_data));
+          } else {
+            int8_t* output_data = output->mutable_data<int8_t>(ctx.GetPlace());
+            dst_memory_p->set_data_handle(to_void_cast<int8_t>(output_data));
+          }
+        } else if (!force_fp32_output) {
+          if (fuse_relu) {
+            uint8_t* output_data = output->mutable_data<uint8_t>(
+                ctx.GetPlace(), ::paddle::memory::Allocator::kDefault,
+                handler->GetDstMemorySize());
+            dst_memory_p->set_data_handle(to_void_cast<uint8_t>(output_data));
+          } else {
+            int8_t* output_data = output->mutable_data<int8_t>(
+                ctx.GetPlace(), ::paddle::memory::Allocator::kDefault,
+                handler->GetDstMemorySize());
+            dst_memory_p->set_data_handle(to_void_cast<int8_t>(output_data));
```

Comment:
Right, it is an option to add such template function. However, it probably makes confusing since FP32 uses the template like you said. So, for INT8, we decide to specify the clear type to describe output memory type of residual block or force_fp32.

Reply:
> Maybe it would be ok to move it to some template function that is parametrized with type of output data? That would make code above much more readable.

I agree with @tpatejko. 

>  it probably makes confusing since FP32 uses the template like you said.

@xiaolil1 Why using template will make confuse?

Reply:
I will optimize it with template as your suggestion.